core/processor/src/main/java/io/quarkus/annotation/processor/ExtensionAnnotationProcessor.java

```diff
@@ -43,6 +43,13 @@ public synchronized void init(ProcessingEnvironment processingEnv) {
 
         boolean useConfigMapping = !Boolean
                 .parseBoolean(utils.processingEnv().getOptions().getOrDefault(Options.LEGACY_CONFIG_ROOT, "false"));
+
+        if (!useConfigMapping) {
+            throw new IllegalArgumentException(
+                    "Starting with Quarkus 3.25, legacy config classes (deprecated since Quarkus 3.19) are not supported anymore. "
+                            + "Please migrate the configuration of your extension to interfaces annotated with @ConfigMapping.");
```

Comment:
```suggestion
                    "Starting with Quarkus 3.25, legacy config classes (deprecated since Quarkus 3.19) are not supported anymore. "
                            + "Please migrate the configuration of your extension to interfaces annotated with @ConfigMapping. See https://quarkus.io/guides/config-mappings#config-mappings for more information.");
```

Reply:
Yeah, I was on the verge of adding a link. Given you thought about it too, it makes sense to add it. I pushed the change.